app/src/main/java/blockeq/com/stellarwallet/activities/BaseActivity.kt

```diff
@@ -21,17 +21,22 @@ open class BaseActivity : AppCompatActivity() {
             val data = WalletApplication.localStore!!.encryptedPhrase
 
             if (data != null && !data.isEmpty()) {
-                launchPINView(data)
+                launchLoginPINView(data)
             }
         }
     }
 
 
     //region Helper Functions
 
-    open fun launchPINView(mnemonic : String) {
+    open fun launchLoginPINView(mnemonic : String) {
         val pinViewState = PinViewState(PinType.CHECK, "", "", mnemonic)
-        PinFlowController.launchPinActivity(this, pinViewState)
+        PinFlowController.launchPinActivity(this, pinViewState, true)
+    }
+
+    open fun launchPINView(pinType: PinType, message: String, mnemonic : String) {
```

Comment:
We could pass in another boolean parameter called `isLogin` and just have one launchPINView function.